### PR TITLE
PLATFORM-1860 Context for image review

### PR DESCRIPTION
**Pull request by RafalWilinski**
## Links

* https://wikia-inc.atlassian.net/browse/PLATFORM-1860

## Description

Image-review service now can also return `context` parameter which tells reviewer where image is located. 

Also, No-Content logic handling has been changed. 

## Reviewers

@vforge @Warkot @bognix

### CODE REVIEW

File: front/common/public/locales/en/image-review.json [hunk 1]
```diff
@@ -5,9 +5,12 @@
     "error-no-access-permissions": "Unauthorized, you don't have permissions to see this page",
     "error-other": "Unable to load image review",
     "error-invalid-data": "Invalid data was returned from Image Review API",
+    "error-no-more-images": "NO MORE IMAGES (╯°□°)╯︵ ┻━┻",
```
Comment (rogatty) [inline front/common/public/locales/en/image-review.json]: Funny, but was it accepted by International team? I just remember what a fuss was there when we used "Map not found. An unicorn is weeping." in maps ;)

File: front/main/app/components/image-review.js [hunk 2]
```diff
@@ -3,12 +3,14 @@ import Ember from 'ember';
...
 
 	actions: {
-		showModal(imageUrl) {
-			this.set('modalImageUrl', imageUrl);
+		showModal(popupModel) {
+			this.set('thumbnailModel', popupModel);
 			this.set('isModalVisible', true);
+		},
+		contextProvided() {
+			return this.get('thumbnailModel').context == "#";
```
Comment (bognix) [inline front/main/app/components/image-review.js]: use `===` and the `'`

File: front/main/app/models/image-review.js [hunk 3]
```diff
@@ -21,7 +21,13 @@ ImageReviewModel.reopenClass({
 				xhrFields: {
 					withCredentials: true
 				},
-				success: (data) => resolve(ImageReviewModel.getImagesAndCount(data.id)),
+				success: (data, textStatus, xhr) => {
+					if (xhr.status == 204) {
```
Comment (bognix) [inline front/main/app/models/image-review.js]: `===` instead of `==`

File: front/main/app/templates/components/image-review.hbs [hunk 4]
```diff
@@ -1,30 +1,39 @@
...
+	{{/modal-dialog}}
 
-<div class="action-buttons">
-    <button {{action (action this.attrs.reviewAndGetMoreImages)}}>{{i18n 'main.review-and-get-more' ns='image-review'}}</button>
-    <button {{action (action this.attrs.getFlaggedOnly)}}>{{i18n 'main.get-flagged' ns='image-review'}}</button>
-</div>
-
-{{alert-notifications alerts=alertNotifications}}
+    <div class="action-buttons">
+        <button {{action (action this.attrs.reviewAndGetMoreImages)}}>{{i18n 'main.review-and-get-more' ns='image-review'}}</button>
```
Comment (bognix) [inline front/main/app/templates/components/image-review.hbs]: do you have to do `{{action (action`+
`{{action attrs. ...` won't be sufficient?

File: front/main/app/templates/image-review.hbs [hunk 5]
```diff
@@ -5,3 +5,4 @@
 	changeItemModel=(action 'changeItemModel')
 	openMainPage=(action 'openMainPage')
 }}
+{{alert-notifications alerts=alertNotifications}}
```
Comment (bognix) [inline front/main/app/templates/image-review.hbs]: do you need it here?

File: front/common/public/locales/en/image-review.json (hunk 1 above)
Comment (RafalWilinski) [inline front/common/public/locales/en/image-review.json]: ┬─┬ノ( º _ ºノ)

Comment (Warkot) [inline front/common/public/locales/en/image-review.json]: Because it should be 'a unicorn'! ლ(ಠ益ಠლ)

**Comment by RafalWilinski** [comment]
@rogatty ?

File: front/main/app/models/image-review.js [hunk 6]
```diff
@@ -21,7 +21,13 @@ ImageReviewModel.reopenClass({
 				xhrFields: {
 					withCredentials: true
 				},
-				success: (data) => resolve(ImageReviewModel.getImagesAndCount(data.id)),
+				success: (data, textStatus, xhr) => {
+					if (xhr.status === 204) {
```
Comment (rogatty) [inline front/main/app/models/image-review.js]: Can you add a comment what 204 means in this situation? Why do we create an empty model instance?

File: front/main/app/templates/components/image-review.hbs (hunk 4 above)
Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: It should be sufficient.

File: front/main/app/templates/components/image-review.hbs [hunk 7]
```diff
@@ -1,30 +1,39 @@
...
+    <p>{{i18n 'main.description' ns='image-review'}}</p>
+    <p>{{i18n 'main.images-to-review-count' ns='image-review'}} <strong>{{model.imagesToReviewCount}}</strong></p>
 
-<ul class="image-review-list">
-	{{#each model.images as |item|}}
-		<li class="image-review-item">
-			{{image-review-item model=item showModal=(action 'showModal')
+    <ul class="image-review-list">
+		{{#each model.images as |item|}}
+            <li class="image-review-item">
```
Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: Please use tabs.

File: front/main/app/templates/components/image-review.hbs [hunk 8]
```diff
@@ -1,30 +1,39 @@
...
+		{{/each}}
+    </ul>
 
-{{#modal-dialog isVisible=isModalVisible}}
-	<img src="{{modalImageUrl}}"/>
-{{/modal-dialog}}
+	{{#modal-dialog isVisible=isModalVisible}}
+        <img src="{{thumbnailModel.fullSizeImageUrl}}"/>
+		{{#if contextProvided}}
+            <a href="{{thumbnailModel.context}}"> {{i18n 'main.image-context' ns='image-review'}} {{thumbnailModel.context}} </a>
```
Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: Formatting is really broken because of tabs/spaces. Just look at it in Github :see_no_evil:

Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: Also, what are the additional spaces inside `<a></a>` needed for?

File: front/main/app/models/image-review.js (hunk 6 above)
Comment (RafalWilinski) [inline front/main/app/models/image-review.js]: Sure, 204 according industry standards means "No content" and that's what exactly it means in this particular situation - no more images available to review. I'm creating empty model just to _instantiate_ `image-review.hbs` into view and show message.

Comment (rogatty) [inline front/main/app/models/image-review.js]: Cool! Can you add it as a comment in the code so the next person won't have to find this PR? Thanks :)

Comment (vforge) [inline front/main/app/models/image-review.js]: @RafalWilinski :+1:

File: front/main/app/components/image-review.js [hunk 9]
```diff
@@ -3,12 +3,14 @@ import Ember from 'ember';
...
 
 	actions: {
-		showModal(imageUrl) {
-			this.set('modalImageUrl', imageUrl);
+		showModal(popupModel) {
+			this.set('thumbnailModel', popupModel);
 			this.set('isModalVisible', true);
+		},
+		contextProvided() {
+			return this.get('thumbnailModel').context === '#';
```
Comment (rogatty) [inline front/main/app/components/image-review.js]: There is `context: image.context || '#'` in the model so I deduce that `'#'` means the context wasn't provided so this condition should inversed. Am I right?

Comment (RafalWilinski) [inline front/main/app/components/image-review.js]: Good catch! Thanks

File: front/main/app/templates/components/image-review.hbs [hunk 10]
```diff
@@ -1,30 +1,39 @@
...
-    <button {{action (action this.attrs.getFlaggedOnly)}}>{{i18n 'main.get-flagged' ns='image-review'}}</button>
-</div>
-
-{{alert-notifications alerts=alertNotifications}}
+    <div class="action-buttons">
+        <button {{action (action this.attrs.reviewAndGetMoreImages)}}>{{i18n 'main.review-and-get-more' ns='image-review'}}</button>
+        <button {{action (action this.attrs.getFlaggedOnly)}}>{{i18n 'main.get-flagged' ns='image-review'}}</button>
+    </div>
+{{else}}
+	<h2> {{i18n 'main.error-no-more-images' ns='image-review'}} </h2>
```
Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: Unnecessary spaces inside `<h2></h2>`.

File: front/main/app/templates/components/image-review.hbs [hunk 11]
```diff
@@ -1,30 +1,39 @@
...
 
-{{#modal-dialog isVisible=isModalVisible}}
-	<img src="{{modalImageUrl}}"/>
-{{/modal-dialog}}
+	{{#modal-dialog isVisible=isModalVisible}}
+        <img src="{{thumbnailModel.fullSizeImageUrl}}"/>
+		{{#if contextProvided}}
+            <a href="{{thumbnailModel.context}}"> {{i18n 'main.image-context' ns='image-review'}} {{thumbnailModel.context}} </a>
+		{{else}}
+            <p> {{i18n 'main.image-context-not-provided' ns='image-review'}} </p>
```
Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: Unnecessary spaces inside `<p></p>`.

File: front/main/app/templates/components/image-review-item.hbs [hunk 12]
```diff
@@ -1,17 +1,17 @@
 <div class="img-container {{model.status}}">
-	<img src="{{thumbnailUrl}}" {{action (action this.attrs.showModal model.fullSizeImageUrl)}}>
-    <form class="button-group">
-        <button class="button-group-item {{if isAccepted 'highlighted' 'inactive'}}"
+	<img src="{{thumbnailUrl}}" {{action (action this.attrs.showModal model)}}>
```
Comment (rogatty) [inline front/main/app/templates/components/image-review-item.hbs]: Same as https://github.com/Wikia/mercury/pull/2084#discussion_r54713166

File: front/main/app/templates/components/image-review.hbs [hunk 13]
```diff
@@ -1,30 +1,39 @@
...
+	{{/modal-dialog}}
 
-<div class="action-buttons">
-    <button {{action (action this.attrs.reviewAndGetMoreImages)}}>{{i18n 'main.review-and-get-more' ns='image-review'}}</button>
-    <button {{action (action this.attrs.getFlaggedOnly)}}>{{i18n 'main.get-flagged' ns='image-review'}}</button>
-</div>
-
-{{alert-notifications alerts=alertNotifications}}
+	<div class="action-buttons">
+		<button {{action (action this.attrs.reviewAndGetMoreImages)}}>{{i18n 'main.review-and-get-more' ns='image-review'}}</button>
```
Comment (rogatty) [inline front/main/app/templates/components/image-review.hbs]: Same as https://github.com/Wikia/mercury/pull/2084#discussion_r54713166

**Comment by rogatty** [comment]
Minor thing (simplifying the code) and it'll LGTM.

**Comment by rogatty** [comment]
LGTM

**Pull request merged**